### DISCUSSION TITLE
I want to import Texture, but I get a compile error

### DISCUSSION THREAD

**Discussion by famibee** [pixijs/pixijs#7394] [Q&A]
Textureをインポートしたいがコンパイルエラーになる

### Current Behavior

- With the description so far ... Compile error【import {Texture} from 'pixi.js';】
  - 今までの記述だと……コンパイルエラー【import {Texture} from 'pixi.js';】

```ts
import {Texture} from 'pixi.js';

const txNormal = new Texture(tx, new Rectangle(0, 0, 100, 40));
```

```console
    ERROR in /Users/***/Documents/MacHD/npm_core/core/src/sn/Button.ts
    ./core/src/sn/Button.ts 220:31-33
    [tsl] ERROR in /Users/***/Documents/MacHD/npm_core/core/src/sn/Button.ts(221,33)
          TS2345: Argument of type 'import("/Users/***/Documents/MacHD/npm_core/node_modules/@pixi/sprite/node_modules/@pixi/core/index").BaseTexture<import("/Users/***/Documents/MacHD/npm_core/node_modules/@pixi/sprite/node_modules/@pixi/core/index").Resource, import("/Users/***/Documents/MacHD2/_Famibee/SKYNo...' is not assignable to parameter of type 'import("/Users/***/Documents/MacHD/npm_core/node_modules/pixi.js/node_modules/@pixi/core/index").BaseTexture<import("/Users/***/Documents/MacHD/npm_core/node_modules/pixi.js/node_modules/@pixi/core/index").Resource, import("/Users/***/Documents/MacHD/npm_co...'.
```
- If you insert a line break in the error message, it will be as follows.
  - エラーメッセージに改行を入れると次のようになる。

```console
    ERROR in /Users/***/Documents/MacHD/npm_core/core/src/sn/Button.ts
    ./core/src/sn/Button.ts 220:31-33
    [tsl] ERROR in /Users/***/Documents/MacHD/npm_core/core/src/sn/Button.ts(220,32)
          TS2345: Argument of type '

import("/Users/***/Documents/MacHD/npm_core/node_modules/
  @pixi/sprite/node_modules/@pixi/core/index").BaseTexture<

import("/Users/***/Documents/MacHD/npm_core/node_modules/
  @pixi/sprite/node_modules/@pixi/core/index").Resource, 

import("/Users/***/Documents/MacHD2/_Famibee/SKYNo...' 
is not assignable to parameter of type '

import("/Users/***/Documents/MacHD/npm_core/node_modules/
  pixi.js/node_modules/@pixi/core/index").BaseTexture<

import("/Users/***/Documents/MacHD/npm_core/node_modules/
  pixi.js/node_modules/@pixi/core/index").Resource, 

import("/Users/***/Documents/MacHD/npm_co...'.
      Types of property 'resource' are incompatible.
        Type '
import("/Users/***/Documents/MacHD/npm_core/node_modules/
  @pixi/sprite/node_modules/@pixi/core/index").Resource' 
is not assignable to type '

import("/Users/***/Documents/MacHD/npm_core/node_modules/
  pixi.js/node_modules/@pixi/core/index").Resource'.

          Property '_width' is protected but type 'Resource' 
is not a class derived from 'Resource'.
```

- If you write import in pixi.js 6.0.2 ...
  - pixi.js 6.0.2 での import 記述をすると……

```ts
import {Texture} from '@pixi/sprite';
```

- I get an "not exported" error.
  - 「エクスポートされてない」エラーになる。

- Other similar phenomena seem to occur in BaseTexture, Resource, Texture, etc.
  - その他 BaseTexture, Resource, Texture などで似たような現象が起こるようです。

### Environment

- **`pixi.js` version**: _6.0.2_
- **Browser & Version**: _VSCode _
バージョン: 1.55.0
コミット: c185983a683d14c396952dd432459097bc7f757f
日付: 2021-03-30T16:01:05.981Z
Electron: 11.3.0
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Darwin x64 20.3.0

- **Running Example**: _url_

**Comment by ShukantPal** [comment]
I don't know Chinese so this answer might be incomplete.

> const txNormal = new Texture(tx, new Rectangle(0, 0, 100, 40));

This seems to be a problem of the typings coming from two packages. Your `tx` seems to have type coming from `/Users/***/Documents/MacHD/npm_core/node_modules/
  @pixi/sprite/node_modules/@pixi/core/index` - but @pixi/sprite should not have a node_modules. This seems like @pixi/sprite hasn't been updated along other PixiJS packages. Can you verify that `node_modules/@pixi/sprite` doesn't have a `node_modules` folder & it's version is correct?

If not, clear your top-level `node_modules` and `package-lock.json`, and reinstall everything.

> import {Texture} from '@pixi/sprite';

Texture isn't exported from @pixi/sprite. You need to import from @pixi/core.

**Reply by famibee** [reply]
Thank you for your reply.
ご回答ありがとうございます。

The point on the last line seems to be my mistake, so I corrected it.
最終行のご指摘は私のミスのようなので、修正してみました。

 ERROR in /...../npm_core/core/src/sn/Button.ts
    ./core/src/sn/Button.ts 9:22-34
    [tsl] ERROR in /...../npm_core/core/src/sn/Button.ts(9,23)
          TS7016: Could not find a declaration file for module '@pixi/core'. '/...../npm_core/node_modules/@pixi/core/lib/core.js' implicitly has an 'any' type.


I tried npm i -D @pixi/core here, but the error is not resolved and ["@pixi/core": "^5.3.9",] and the old version are installed.
Was this an unnecessary installation in the first place?
ここで npm i -D @pixi/core をしてみましたが、エラーは解消されず【"@pixi/core": "^5.3.9",】と古いバージョンがインストールされます。
そもそもこれは不要なインストールだったでしょうか？

**Comment by ShukantPal** [comment]
If you’re using the pixi.js bundle, you should not be installing most @pixi/ scoped packages directly. Make sure that you’re using PixiJS 6.0.0+ packages for proper typings.

**Reply by famibee** [reply]
I made a reproduction sample.
再現サンプルを作りました。

(pixi6tst.zip)
https://firestorage.jp/download/34f6292396c6526dfb9a61f57e766bc877b300f1

1. Unzip and open the folder with VS Code etc.
2. You should see two pigs in [npm i] and [task start]. Up to this point, it is a normal case operation.
3. [npm i -S pix i-sound] and [task start]will cause an error. (TS2345: Argument of type'import ("...)
4. Rebuilding with [task rebuild] does not resolve the error.
5. If you use the old version with [npm un -S pixi.js && npm i -S pixi.js@5.3.9], the error disappears.
6. [npm un -S pixi.js && npm i -S pixi.js@latest] An error will occur if you update to the new version.

1.解凍してVSCodeなどでフォルダを開いて下さい。
2.【npm i】【task start】で豚ちゃんが二匹表示されると思います。ここまでは正常ケースの動作です。
3.【npm i -S pixi-sound】【task start】するとエラーが発生します。（TS2345: Argument of type 'import("...）
4.【task rebuild】でリビルドしてもエラーは解消されません。
5. 【npm un -S pixi.js && npm i -S pixi.js@5.3.9】と、旧版にするとエラーが消えます。
6. 【npm un -S pixi.js && npm i -S pixi.js@latest】新版にするとエラーになります。


The combination of pixi-sound and pixi.js@6.x.x seems to be a problem.
　pixi-sound と pixi.js@6.x.x の組み合わせで不具合になるようです。

**Reply by bigtimebuddy** [reply]
Please try using `@pixi/sound` [see release](https://github.com/pixijs/sound/releases/tag/v4.0.0).

**Reply by famibee** [reply]
Thank you and I'm sorry, I solved it.
ありがとう、そしてすみません、解決しました。
